src/profile/v100/app_preferences/resource_preferences/resource_preference_kind.rs

```diff
@@ -0,0 +1,10 @@
+use crate::prelude::*;
+
+#[derive(
+    Clone, Debug, PartialEq, Eq, Hash, derive_more::Display, uniffi::Enum,
+)]
+pub enum ResourcePreferenceKind {
```

Comment:
maybe `enum AssetAddress` ?

Reply:
there is no such thing as a `FungibleAddress` (so it would represent fungibles as `resource`), nor we want a `NonFungibleResourceAddress` (we want `NonFungibleGlobalId`), so not sure this address_union! macro helps us here.